### PR TITLE
I18nify TableStylingWorkflow and its related components

### PR DESCRIPTION
**Pull request by ykiu**
### What this PR does

I18nify TableStylingWorkflow and its related components. This pull request primarily focuses on replacing hardcoded strings with i18next.t() calls. It does not modify any other aspects of the codebase, but a few changes have been made to the IDs of certain dimensions to (a) enable translation, (b) enhance consistency, (c) improve readability and (d) resolve React warnings related to "two children with the same key":

(a) Some dimensions previously had only IDs, such as "Style options", which were visible in the user interface. In this PR, such dimensions had their IDs changed to kebab-case and got explicit display names.

(b) Certain IDs were written in mixedCase, while the majority were in kebab-case. In cases where it was deemed safe, mixedCase IDs have been converted to kebab-case.

(c) Some IDs included the abbreviation "col", which could refer to color or column depending on the context. To enhance clarity, these abbreviations have been expanded to their full forms. In addition, it appeared that `${id}-lead-power` actually meant lead "time", so that has also been corrected.

(d) Duplicate IDs were identified, leading to React warnings about "two children with the same key". To resolve this issue, unique IDs have been assigned.

I am pretty sure that these changes do not impact the functionality of the application, but here I include a list of the modified IDs for reference:

| main                      | HEAD                         |  Reason   |
| ------------------------- | ---------------------------- | --- |
| Fill Color                | fill                         | a    |
| table-color-col           | table-color-column           | c    |
| Type                      | type                         | a    |
| Scheme                    | scheme                       | a    |
| numberOfBins              | number-of-bins               | b    |
| Display range             | display-range                | a    |
| Bins                      | bins                         | a    |
| bin-${idx}-col            | bin-${idx}-color             | c    |
| Colors                    | colors                       | a    |
| enum-${idx}-col           | enum-${idx}-color            | c    |
| null-col                  | null-color                   | c    |
| outlier-col               | outlier-color                | c    |
| Additional colors         | additional-colors            | a    |
| region-col                | region-color                 | c    |
| null-col                  | null-color                   | c    |
| outlier-col               | outlier-color                | c    |
| Point size                | point-size                   | a    |
| Region mapping            | region-mapping               | a    |
| Legend                    | legend                       | a    |
| Style options             | style-options                | a    |
| table-style               | longitude-column             | d    |
| table-style               | latitude-column              | d    |
| Time options              | time-options                 | a    |
| table-time-col            | table-time-column            | c    |
| table-end-time-col        | table-end-time-column        | c    |
| table-time-id-cols        | table-time-id-columns        | c    |
| Workbench options         | workbench-options            | a    |
| showDisableStyleOption    | show-disable-style-option    | b    |
| showDisableTimeOption     | show-disable-time-option     | b    |
| enableManualRegionMapping | enable-manual-region-mapping | b    |
| Variable/column           | variable-and-column          | a    |
| ${id}-lead-power          | ${id}-lead-time              | c    |
| Trail style options       | trail-style-options          | a    |

Additionally, Japanese translations have been added to validate the effectiveness of these changes in ensuring the components are translatable.

## Test me

Build with TerriaMap and confirm that the style UI remains functional.

In addition, verify that the UI is also available in other languages. As there's only Japanese translations for now, update `languageConfiguration` in config.json like the following and reload TerriaMap:

```json
"languageConfiguration": {
      "enabled": true,
      "debug": false,
      "languages": {
            "en": "English",
            "fr": "Français",
            "ja": "Japanese"
      },
      "fallbackLanguage": "en"
},
```

Please note that even after switching the language by using the language switcher UI, some strings may remain untranslated. To see all translations, reload TerriaMap once again.

### Checklist

- [ ] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
- [ ] I've updated relevant documentation in `doc/`.
- [x] I've updated CHANGES.md with what I changed.
- [x] I've provided instructions in the PR description on how to test this PR.

### CODE REVIEW

File: lib/Models/Workflows/TableStylingWorkflow.ts [hunk 1]
```diff
@@ -513,7 +519,10 @@ export default class TableStylingWorkflow
         {
           type: "select",
           id: "table-style",
-          name: "Style",
+
+          name: i18next.t(
+            "models.tableStyling.data.selectableDimensions.tableStyle.dataset"
```
Comment (nf-s) [inline lib/Models/Workflows/TableStylingWorkflow.ts]: `models.tableStyling.data.selectableDimensions.tableStyle.name`

File: lib/Models/Workflows/TableStylingWorkflow.ts [hunk 2]
```diff
@@ -1914,14 +2278,17 @@ export default class TableStylingWorkflow
...
       "outline",
       this.tableStyle.outlineStyleMap,
       (id, outlineTraits, nullValues) => [
         {
           type: "color",
           id: `${id}-color`,
-          name: `Color`,
+
+          name: i18next.t(
+            "models.tableStyling.outline.selectableDimensions.color"
```
Comment (nf-s) [inline lib/Models/Workflows/TableStylingWorkflow.ts]: `models.tableStyling.outline.selectableDimensions.color.name`

File: lib/Models/Workflows/TableStylingWorkflow.ts [hunk 3]
```diff
@@ -1931,7 +2298,10 @@ export default class TableStylingWorkflow
         {
           type: "numeric",
           id: `${id}-width`,
-          name: "Width",
+
+          name: i18next.t(
+            "models.tableStyling.outline.selectableDimensions.width"
```
Comment (nf-s) [inline lib/Models/Workflows/TableStylingWorkflow.ts]: `models.tableStyling.outline.selectableDimensions.width.name`

**Review by nf-s — CHANGES_REQUESTED** [review]
Hi @ykiu

Thank you so much for making this contribution, this is awesome!

I have three small comments on translation strings that slightly deviate from your pattern.

Other than that, I am happy to merge it in!

Thanks again 🙂

**Comment by ykiu** [comment]
Hi @nf-s, thanks for your review and good catch!

Being able to style things via the UI is awesome, by the way. Thank you for your hard work!

**Review by nf-s — APPROVED** [review]
No problem, thank you for using the feature 🙂 

Thanks again for the PR!

**Pull request merged**